app/javascript/packages/document-capture/hooks/use-i18n.js

```diff
@@ -46,17 +46,36 @@ export function formatHTML(html, handlers) {
   return parts.filter(Boolean);
 }
 
+/**
+ * Returns string with variable substitution.
+ *
+ * @param {string} string Original string.
+ * @param {Record<string,string>} variables Variables to replace.
+ *
+ * @return {string} String with variables substituted.
+ */
+export function replaceVariables(string, variables) {
+  return Object.keys(variables).reduce(
+    (result, key) => result.replace(new RegExp(`%{${key}}`, 'g'), variables[key]),
+    string,
+  );
+}
+
 function useI18n() {
   const strings = useContext(I18nContext);
 
   /**
    * Returns the translated string by the given key.
    *
    * @param {string} key Key to retrieve.
+   * @param {Record<string,string>=} variables Variables to substitute in string.
    *
    * @return {string} Translated string.
    */
-  const t = (key) => (Object.prototype.hasOwnProperty.call(strings, key) ? strings[key] : key);
+  function t(key, variables) {
```

Comment:
should we make `variables` an optional value?
```suggestion
  function t(key, variables = {}) {
```

Reply:
> should we make `variables` an optional value?

I can see some benefit, since it would avoid the ternary expression in this function.

My thinking was: If we are given no variables, then we should know to pass back the string as-is. While it's true that `replaceVariables` with an empty object has the same result, I've a slight preference with this implementation as being more representative of this written expectation.

Reply:
Ok, that works for me